extensions/objects/templates/literally-canvas-diagram-editor.component.ts

```diff
@@ -65,30 +93,157 @@ angular.module('oppia').component('literallyCanvasDiagramEditor', {
           'px X ' + maxHeight + 'px to ensure that it fits in the card.');
       };
 
-      ctrl.validate = function(data) {
-        // TODO(#9357): Will be implemented once the svg data saving
-        // functionality is implemented.
+      ctrl.isDiagramCreated = function() {
+        // This function checks if any shape has been created or not.
+        if (typeof ctrl.lc !== 'undefined' &&
+          ctrl.diagramStatus === 'editing') {
+          var svgString = ctrl.lc.getSVGString();
+          var domParser = new DOMParser();
+          var doc = domParser.parseFromString(svgString, 'text/xml');
+          var elements = doc.querySelectorAll('svg > g > *');
+          if (elements.length > 0) {
+            return true;
+          }
+        }
         return false;
       };
 
-      ctrl.$onInit = function() {
-        LC.defineSVGRenderer(
-          'Rectangle', LiterallyCanvasHelperService.rectangleSVGRenderer);
-        LC.defineSVGRenderer(
-          'Ellipse', LiterallyCanvasHelperService.ellipseSVGRenderer);
-        LC.defineSVGRenderer(
-          'Line', LiterallyCanvasHelperService.lineSVGRenderer);
-        LC.defineSVGRenderer(
-          'LinePath', LiterallyCanvasHelperService.linepathSVGRenderer);
-        LC.defineSVGRenderer(
-          'Polygon', LiterallyCanvasHelperService.polygonSVGRenderer);
-        LC.defineSVGRenderer(
-          'Text', LiterallyCanvasHelperService.textSVGRenderer);
+      ctrl.isUserDrawing = function() {
+        if (typeof ctrl.lc !== 'undefined' &&
+           ctrl.lc._shapesInProgress.length > 0) {
+          return true;
+        }
+        return false;
+      };
+
+      var getTrustedResourceUrlForSVGFileName = function(svgFileName) {
+        var encodedFilepath = window.encodeURIComponent(svgFileName);
+        return $sce.trustAsResourceUrl(
+          AssetsBackendApiService.getImageUrlForPreview(
+            ctrl.entityType, ctrl.entityId, encodedFilepath));
+      };
+
+      ctrl.setSavedSVGFilename = function(filename, setData) {
+        ctrl.diagramStatus = 'saved';
+        ctrl.data = {
+          savedSVGFileName: filename,
+          savedSVGUrl: getTrustedResourceUrlForSVGFileName(filename)
+        };
+        ctrl.value = filename;
+        if (setData) {
+          $http.get(ctrl.data.savedSVGUrl).then(function(response) {
+            ctrl.savedSVGDiagram = response.data;
+          });
+        }
+      };
+
+      ctrl.postSVGToServer = function(dimensions, resampledFile) {
```

Comment:
Hmm, not sure if you need to do it in this PR, but since this is an RTE element, you'd need to support local storage as well so that even in the 'create' modals of questions and skills, this can be created.
You can check the implementation of 'ImageLocalStorageService' in filepath-editor.directive.ts for reference.

Reply:
Done.